src/mode/modeHandler.ts

```diff
@@ -1510,7 +1510,17 @@ export class ModeHandler implements vscode.Disposable {
       opCursorCharDecorations
     );
 
-    // TODO: draw marks (#3963)
+    if (configuration.showMarksInGutter) {
```

Comment:
If this is `false`, we should probably clear the decorations by calling `editor.setDecorations(decorationType, [])` for each `decorationType` we've constructed (this will only matter if the user disables it at runtime).